homeassistant/components/zeroconf/__init__.py

```diff
@@ -379,11 +381,38 @@ async def async_setup(self) -> None:
             self.zeroconf, types, handlers=[self.async_service_update]
         )
 
+        async_dispatcher_connect(
+            self.hass,
+            config_entries.SIGNAL_CONFIG_ENTRY_CHANGED,
+            self._handle_config_entry_changed,
+        )
+
     async def async_stop(self) -> None:
         """Cancel the service browser and stop processing the queue."""
         if self.async_service_browser:
             await self.async_service_browser.async_cancel()
 
+    @callback
+    def _handle_config_entry_changed(
+        self,
+        change: config_entries.ConfigEntryChange,
+        entry: config_entries.ConfigEntry,
+    ) -> None:
+        """Handle config entry changes."""
+        if (
+            change != config_entries.ConfigEntryChange.REMOVED
+            or entry.source != config_entries.SOURCE_IGNORE
+            or not (discovery_keys := entry.data.get("discovery_keys"))
+        ):
+            return
+        for discovery_key in discovery_keys:
+            if discovery_key["domain"] != DOMAIN or discovery_key["version"] != 1:
+                continue
+            _type = discovery_key["key"][0]
+            name = discovery_key["key"][1]
+            _LOGGER.debug("Rediscover unignored service %s.%s", _type, name)
+            self._async_service_update(self.zeroconf, _type, name)
```

Comment:
If `zeroconf` is an instance attribute we don't need to pass it around to instance methods.

Reply:
You're right, but the zeroconf instance is passed to us by the zeroconf callback, I didn't want to change the existing pattern. I can clean it up though?

Reply:
Follow up is ok too.